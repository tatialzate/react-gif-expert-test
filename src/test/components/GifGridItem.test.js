import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe( 'Pruebas componente GifGridItem', () => {

    const img = {
        url: 'https://media3.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=78a8ec572354v6c7c7njjwlqwua9hdik9nxxdcc23a3l3ogk&rid=giphy.gif&ct=g',
        title: 'saitama GIF'
    }
    const wrapper = shallow( <GifGridItem {...img}/> );

    test( 'Debe mostrarse el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test( 'Debe presentar en el pÃ¡rrafo el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim()).toBe( img.title );
    })

    test( 'Debe presentar la url en el src y el title en el alt', () => {
        const image = wrapper.find('img');
        expect( image.prop('src')).toBe( img.url );
        expect( image.prop('alt')).toBe( img.title );
    })

    test( 'El div debe contener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn')).toBe(true);
    })
})