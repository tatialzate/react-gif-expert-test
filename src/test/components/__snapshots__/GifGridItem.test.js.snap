// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente GifGridItem Debe mostrarse el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="saitama GIF"
    src="https://media3.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=78a8ec572354v6c7c7njjwlqwua9hdik9nxxdcc23a3l3ogk&rid=giphy.gif&ct=g"
  />
  <p>
     
    saitama GIF
     
  </p>
</div>
`;
